corona-warn-app/cwa-app-android
# Refactor riskcard & states; Chapter1 (EXPOSUREAPP-4049, EXPOSUREAPP-3834)

* Removed `RiskLevelConstants` which we not useful anymore and just unnecessary complicated
* Simplified `RiskLevel` to `RiskState` which just has 3 states now `INCREASED_RISK`, `LOW_RISK`, `FAILED`
* Simplified state behavior of the risk cards on the homescreen.
    * Disabled tracing overrides all other cards, as it's a current state we want the user to fix
    * Otherwise if the last calculation failed (for any reason), we show the "tracing failed" card
    * If the last calculation did not fail we display either a high or low risk card

## TODO
* [x] No data yet, fresh install, airplane mode, manually press "restart", card seems to be just empty (light mode)
* [x] No data yet, fresh install, airplane mode, manually press "restart", card seems to be just empty, now enable WIFI, card is empty then green card appears (light mode)
* [x] Tracing disabled, light mode, divider is missing or invisible
* [x] Missing "contacts" when displaying low risk card
* [x] Negative days when time traveling backwards
* [x] Remove "minimum tracing time active" restrictions which should be removed according to EXPOSUREAPP-3051 (moved to extra PR #1762)

## Testing
* Make sure all 4 cards look correct (compare with figma, though it's not 100% current)
   * Red card (high rish)
   * Green card (low risk)
   * Tracing stopped
   * Tracing failed
* Check all designs in dark mode too
